src/connector/src/source/nexmark/source/combined_event.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use nexmark::event::{Auction, Bid, Person};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct CombinedEvent {
```

Comment:
Why not directly use `nexmark::Event`? It can also be serialized into json. 🤔

Reply:
My intention is to have a struct 1-1 mapped to the type definitions in `CREATE SOURCE nexmark`.
So I want to have an additional field `event_type` which is serialized to an integer.
And also for each unified data event, i.e. `CombinedEvent`, only one of the three [nexmark::Event](https://docs.rs/nexmark/latest/nexmark/event/enum.Event.html)s can be non-null.
In JSON, roughly: `{"event_type":0, "person": {XXX}, "bid": null, "auction": null}`

Do you mean by using `nexmark::Event` directly, it will also be serialized into `{"person": {XXX}, "bid": null, "auction": null}`? 🤔

Reply:
Okay I got it.
`nexmark::Event` will be serialized into sth like `{"Person": {XXX}}`.